### DISCUSSION TITLE
Can  nats.go  support custom authentication method

### DISCUSSION THREAD

**Discussion by peaka9** [nats-io/nats.go#766] [Q&A]
Hello,
    If we want to use our own Username/Password authorization,  what could we do? Does it support custom authentication method?

**Comment by wallyqs** [comment]
There is [nats.UserInfo](https://pkg.go.dev/github.com/nats-io/go-nats#UserInfo) option to username/password in the client to authenticate with the server: https://docs.nats.io/developing-with-nats/security/userpass
